extensions/zoom/src/components/CreateMeetingForm.tsx

```diff
@@ -17,103 +17,92 @@ type MeetingFormProps = {
-        duration: duration ? parseInt(duration) : 60,
-        timezone: Intl.DateTimeFormat().resolvedOptions().timeZone,
-      };
-
-      const meeting = await createScheduledMeeting(payload);
-
-      toast.style = Toast.Style.Success;
-      toast.title = "Scheduled meeting";
-
-      toast.primaryAction = {
-        title: "Open Meeting",
-        shortcut: { modifiers: ["cmd", "shift"], key: "o" },
-        onAction: () => {
-          open(meeting.join_url);
-          return toast.hide();
-        },
-      };
-
-      toast.secondaryAction = {
-        title: "Copy Join URL",
-        shortcut: { modifiers: ["cmd", "shift"], key: "c" },
-        onAction: () => {
-          Clipboard.copy(meeting.join_url);
-          toast.title = "Copied to clipboard";
-          toast.message = meeting.join_url;
-        },
-      };
-
-      setStartTime(undefined);
-      setDuration("");
-      setTopic("");
-      setAgenda("");
-
-      startTimeRef.current?.focus();
-    } catch (error) {
-      toast.style = Toast.Style.Failure;
-      toast.title = "Failed to schedule meeting";
-      toast.message = getErrorMessage(error);
-    }
-  }
+  const { handleSubmit, itemProps, focus, reset } = useForm<MeetingFormValues>({
+    async onSubmit(values) {
+      const toast = new Toast({ style: Toast.Style.Animated, title: "Scheduling meeting" });
+      await toast.show();
+
+      try {
+        const payload = {
+          topic: values.topic,
+          agenda: values.agenda,
+          start_time: format(values.start_time, "yyyy-MM-dd'T'HH:mm:ss"),
+          duration: values.duration ? parseInt(values.duration) : 60,
+          timezone: Intl.DateTimeFormat().resolvedOptions().timeZone,
+        };
+
+        const meeting = await createScheduledMeeting(payload);
+
+        toast.style = Toast.Style.Success;
+        toast.title = "Scheduled meeting";
+
+        toast.primaryAction = {
+          title: "Open Meeting",
+          shortcut: { modifiers: ["cmd", "shift"], key: "o" },
+          onAction: () => {
+            open(meeting.join_url);
+            return toast.hide();
+          },
+        };
+
+        toast.secondaryAction = {
+          title: "Copy Join URL",
+          shortcut: { modifiers: ["cmd", "shift"], key: "c" },
+          onAction: () => {
+            Clipboard.copy(meeting.join_url);
+            toast.title = "Copied to clipboard";
+            toast.message = meeting.join_url;
+          },
+        };
+
+        reset({
+          start_time: undefined,
+          duration: "",
+          topic: "",
+          agenda: "",
+        });
+
+        focus("start_time");
+      } catch (error) {
+        toast.style = Toast.Style.Failure;
+        toast.title = "Failed to schedule meeting";
+        toast.message = getErrorMessage(error);
+      }
+    },
+    validation: {
+      start_time: FormValidation.Required,
+      duration: (value) => {
+        if (isNaN(parseInt(value ?? ""))) {
```

Comment:
this isn't required, right?
```suggestion
        if (value && isNaN(parseInt(value ?? ""))) {
```

Reply:
Indeed, the duration isn't required.

Reply:
right so the check won't work because `isNaN(parseInt('')) === true`

Reply:
Oh, yes indeed. Thanks for pointing it out.